sdk/core/azure-core/src/http/user_agent.cpp

```diff
@@ -166,8 +165,8 @@ namespace Azure { namespace Core { namespace Http { namespace _detail {
 
     static std::string const osVer = GetOSVersion();
     telemetryId << "azsdk-cpp-" << componentName << "/" << componentVersion << " (" << osVer << " "
-                << "Cpp/" << cplusplusValue << ")";
+                << "Cpp/" << __cplusplus << ")";
```

Comment:
This will be `201402L` every single time, because it is the value of `__cplusplus` at the time of compiling the SDK, which we know already - in cmake files we do `set(CMAKE_CXX_STANDARD 14)`.

This will say nothing about which C++ standard version our users have set when they *consume* the SDK.

Reply:
Exactly - the stuff that lives in `.hpp` files gets built within the context of compiler settings of the code that includes the file.

Reply:
Gotcha.
I just tested in a sample, where I bumped the C++ standard version, and it had no impact on the user-agent string, as you suspected.
> std::cout << UserAgentGenerator::GenerateUserAgent("storage-blob", "11.0.0", "") << std::endl;

Moving the use of the macro to `.hpp` fixed it. Ty!